rfcs/resource-prototypes.md

```diff
@@ -0,0 +1,237 @@
+
+## Posts [/posts]
+
+### List posts [GET]
+
++ Response 200
+    + Attributes (array[Post], required, fixed-type)
+
++ Response 429 (application/json)
+    + Attributes
+        + status: tooManyRequests (string, required, fixed)
+        + waitFor: `10` (number, required) - wait before next request
+
++ Response 400 (application/json)
+    + Attributes
+        + status: badRequest (string, required, fixed)
+        + errors (array[string], required, fixed-type) - list of errors in request
+
+### Show post [GET]
+
++ Response 200
+    + Attributes (Post, required)
+
++ Response 429 (application/json)
+    + Attributes
+        + status: tooManyRequests (string, required, fixed)
+        + waitFor: `10` (number, required) - wait before next request
+
++ Response 400 (application/json)
+    + Attributes
+        + status: badRequest (string, required, fixed)
+        + errors (array[string], required, fixed-type) - list of errors in request
+
+## Users [/users]
+
+### List users [GET]
+
++ Response 200
+    + Attributes (array[User], required, fixed-type)
+
++ Response 429 (application/json)
+    + Attributes
+        + status: tooManyRequests (string, required, fixed)
+        + waitFor: `10` (number, required) - wait before next request
+
++ Response 400 (application/json)
+    + Attributes
+        + status: badRequest (string, required, fixed)
+        + errors (array[string], required, fixed-type) - list of errors in request
+
++ Response 403 (application/json)
+    + Attributes
+        + status: forbidden (string, required, fixed)
+
+### Create user [POST]
+
++ Response 200
+    + Attributes (User, required)
+
++ Response 429 (application/json)
+    + Attributes
+        + status: tooManyRequests (string, required, fixed)
+        + waitFor: `10` (number, required) - wait before next request
+
++ Response 400 (application/json)
+    + Attributes
+        + status: badRequest (string, required, fixed)
+        + errors (array[string], required, fixed-type) - list of errors in request
+
++ Response 403 (application/json)
+    + Attributes
+        + status: forbidden (string, required, fixed)
+```
+
+As you can see responses 429, 400 and 403 are duplicated several times. It is
+possible to extract response description into
+[Data Structures](https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#def-data-structures)
+section and it will help to reduce duplication but you have to write something
+like
+
+```apib
++ Response 429 (application/json)
+    + Attributes (TooManyRequestsError)
+```
+
+for every resource. It is possible to remove duplication completely using new
+proposed syntax:
+
+```apib
+# Data Structures
+
+## Post
+
++ title: `test` (string, required)
++ body: `hello world` (string, required)
+
+## User
+
++ name: `John Smith` (string, required)
++ email: `user@example.com` (string, required)
+
+## Authorized (Resource Prototype)
+
++ Response 403 (application/json)
```

Comment:
@kylef I've created separate section for resource prototypes. Structure of resource prototype differs a lot from mson data structures, so it will be better for user to have resource prototypes in separate section because in this case API will be more readable and clear.